### DISCUSSION TITLE
Target and attack

### DISCUSSION THREAD

**Discussion by Conan1996-2** [trickerer/Trinity-Bots#371] [Ideas]
Love the Module, just installed for azerothcore and I have been playing with it.

Couple of things:

Can we have a command to send a specific bot to attack, like:
- .npcb attack Bromos mytarget
This way, if I play a healer or ranged, I can create a one button macro that will send the Tank (Bromos in this case) in to start the attack once Im ready for the fight. He would run into position, gain agro and do his thing while im out of range... then i can come into range and do my thing. The current way is for me to run in and then once they are in range, start attacking and move backwards while he tries to gain agro.

Another, more tough idea. have the ability to create waypoints, delay timers for individual bots, so they look more lively. As an example, I could take Bromos (again, same bot) and stand in front of the auctioneer, start a new path for the bot, then Record the position with a set delay of, 10 seconds. Then move to another position and set that (Mailbox?) and set that as a new waypoint, with a delay of 5 seconds, then go to the bank and set another waypoint with a 10 second delay... see where im going with this? Once I finish, the bot will travel the path and back to the begining when he has no master.

**Comment by trickerer** [comment]
1. This could be implemented via new `.npcbot order ...` command I think. But this will only affect aggroing. Targeting priority is managed through group icons and formation gossip menu options
2. Waypoints aren't possible right now, let alone custom ones. Related: #339

**Reply by Conan1996-2** [reply]
This was a fast response. again thank you for the module. I have used the .npcbot order, but this is to cast a spell. In Bromos case (a Paladin), I would cast Avengin Shield, issue is, he doesnt move close enough to cast, and this is npc specific spell.

**Reply by trickerer** [reply]
Of course it doesn't always work. I meant a new `.npcbot order ...` subcommand. Although cast order kinda works for tanks with 30yds taunt
